### ISSUE TITLE
[BUG] no such attribute error when evaluating an expression that should have the attribute

### ISSUE THREAD

**Issue by timostamm** [bufbuild/protovalidate-go#296]
## Description

When validating a custom predefined rule, I ran into an unexpected error:

> runtime error: error evaluating field_mask.fm_rules_extension: no such attribute(s): rule

Here's the rule:

```proto
syntax = "proto2";
package tests.example.v1;
import "buf/validate/validate.proto";
import "google/protobuf/field_mask.proto";

message ExtendedFieldMaskRules {
  optional google.protobuf.FieldMask fm = 1 [(buf.validate.field).field_mask.(fm_rules_extension) = {
    paths: ["a"]
  }];
}
extend buf.validate.FieldMaskRules {
  optional google.protobuf.FieldMask fm_rules_extension = 1801 [(buf.validate.predefined).cel = {
    id: "field_mask.fm_rules_extension"
    expression: "this == rule"
  }];
}
```

The Go test code:

```go
	msg := &pb.ExtendedFieldMaskRules{
		Fm: &fieldmaskpb.FieldMask{
			Paths: []string{"a"},
		},
	}
	err = val.Validate(msg)
	require.NoError(t, err)
```

In the CEL expression, `rule` should be available.

## Environment

cel-go 5439f0512d2aa9585e6251badb2f4912b6deafad

## Additional Context

We fixed a related issue in https://github.com/bufbuild/protovalidate-go/pull/293, but it looks like the issue still persists for custom predefined rules.
